var/spack/repos/builtin/packages/rclone/package.py

```diff
@@ -17,6 +17,10 @@ class Rclone(Package):
 
     depends_on("go", type='build')
 
+    def setup_environment(self, spack_env, run_env):
+        # Point GOPATH at the top of the staging dir for the build step.
+        spack_env.prepend_path('GOPATH', self.stage.path)
```

Comment:
would this variable be better set to a cache path that spack already has?

Reply:
This is essentially the stage path-based approach -- using the standard Spack staging directory -- used in the custom Singularity installation process.

Reply:
fair enough, I think at somepoint spack will want to take advantage of go's proxying support to allow for air gapped builds the go packaging world of dep management right how is hit and miss.

Reply:
@jcftang -- We kicked this around a bit when a similar change was made in the singularity package: https://github.com/spack/spack/pull/11662#discussion_r292256135

There's definitely an opportunity to figure out what the right thing is going to look like....

Reply:
@hartzell, @jcftang   I'm open to suggestions since I don't have experience (yet) with `go`.

Reply:
@hartzell sounds good that others have somewhat discussed this already a bit, I'd agree that in the future there's needs to be some thought put into the go based spack packages.

@tldahlgren this looks good to me if someone else approves the PR to merge